source/extensions/tracers/xray/tracer.h

```diff
@@ -1,19 +1,83 @@
 #pragma once
 
 #include <string>
+#include <utility>
+#include <vector>
 
 #include "envoy/common/time.h"
 #include "envoy/tracing/http_tracer.h"
 
 #include "extensions/tracers/xray/sampling_strategy.h"
+#include "extensions/tracers/xray/xray_configuration.h"
 
+#include "absl/container/flat_hash_map.h"
 #include "absl/strings/string_view.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace Tracers {
 namespace XRay {
 
+static const char XRayTraceHeader[] = ":x-amzn-trace-id";
```

Comment:
This shouldn't begin with a colon.   It's not a required HTTP/2 header.

See Section 8.1.2.3 in this RFC for more information https://tools.ietf.org/html/rfc7540#section-8.1.2

Sorry if this wasn't clear in our earlier conversation.

Reply:
What I understood from you is that Envoy will change it to H2 header (adding the colon). Let's talk offline